packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -160,6 +160,20 @@ export async function fetchServerResponse(
         : 'low'
       : 'auto'
 
+    if (process.env.NODE_ENV === 'production') {
+      url = new URL(url)
+      if (process.env.__NEXT_CONFIG_OUTPUT === 'export') {
```

Comment:
I moved this into a less generic path so it only applies to navigations, not per-segment prefetches.